src/utils/createContextWrapper.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+/**
+ * Creates new trigger class that injects context into overlay.
+ */
+export default function createContextWrapper(Trigger, propName) {
+  return function (contextTypes) {
+    class ContextWrapper extends React.Component {
+      getChildContext() {
+        return this.props.context;
+      }
+
+      render() {
+        // Strip injected props from below.
+        const {wrapped, ...props} = this.props;
+        delete props.context;
```

Comment:
I see, you're right that `this.props` is not being mutated, my bad. @AlexKVal I think this should be ok here.